http_client.go

```diff
@@ -10,6 +10,8 @@ import (
 
 var client = new(http.Client)
 
+type resourceResponse interface{}
```

Comment:
what is this? Shouldn't use `interface{}` without a good reason--what's the reasoning here?